server/src/typeDefs/article.ts

```diff
@@ -0,0 +1,15 @@
+import { gql } from 'apollo-server'
+
+export const article = gql`
+  type Article {
+		id:           String!
+		createdAt:    DateTime!
+		updatedAt:    DateTime!
+		sourceId:    String!
+		author:       String
+		title:        String!
+		slug:         String!
+		shortContent: String!
+		users:        [ArticlesForUser]
```

Comment:
so unfortunately i think this idea of "how is the api different from the database" is still missing some understanding from you guys.

here you can simply put "isFavorite" "isRead" directly on the `type Article`, there is no need for `type ArticlesForUser`. in fact, this whole line:
```
		users:        [ArticlesForUser]
```

is mega confusing. try rereading it now again and seeing if the name makes sense.